src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,241 @@
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'lodash';
+import Button from './Button';
+import * as Expensicons from './Icon/Expensicons';
+import styles from '../styles/styles';
+import AttachmentView from './AttachmentView';
+import SwipeableView from './SwipeableView';
+import addEncryptedAuthTokenToURL from '../libs/addEncryptedAuthTokenToURL';
+import themeColors from '../styles/themes/default';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import canUseTouchScreen from '../libs/canUseTouchscreen';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onArrowPress: PropTypes.func,
+};
+
+const defaultProps = {
+    sourceURL: '',
+    reportActions: {},
+    onArrowPress: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.makeAndSetArrowState = this.makeAndSetArrowState.bind(this);
+        this.onPressShowArrow = this.onPressShowArrow.bind(this);
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+        this.handleArrowPress = this.handleArrowPress.bind(this);
+        this.onShowArrow = this.onShowArrow.bind(this);
+
+        this.state = {showArrows: this.canUseTouchScreen};
+    }
+
+    componentDidMount() {
+        this.makeAndSetArrowState();
+        if (this.canUseTouchScreen) {
+            return;
+        }
+        document.addEventListener('keydown', this.handleArrowPress);
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeAndSetArrowState();
+    }
+
+    componentWillUnmount() {
+        if (this.canUseTouchScreen) {
+            return;
+        }
+        document.removeEventListener('keydown', this.handleArrowPress);
+    }
+
+    /**
+     * Toggles the visibility of the arrows on mouse hover
+     * @param {Boolean} showArrows
+     */
+    onShowArrow(showArrows) {
+        this.setState({showArrows});
+    }
+
+    /**
+     * Helper to provide a check for touchscreens
+     */
+    onPressShowArrow() {
+        if (!this.canUseTouchScreen) {
+            return;
+        }
+        this.onShowArrow(!this.state.showArrows);
+    }
```

Comment:
It would be better if we separate base, native & web platform codes separately.

Reply:
This shouldn't be an issue